protocols/kad/src/protocol.rs

```diff
@@ -142,21 +143,21 @@ impl From<KadPeer> for proto::message::Peer {
 //       `OutboundUpgrade` to be just a single message
 #[derive(Debug, Clone)]
 pub struct KademliaProtocolConfig {
-    protocol_name: Cow<'static, [u8]>,
+    protocol_names: SmallVec<[Cow<'static, [u8]>; 2]>,
```

Comment:
Why use a `SmallVec` instead of a `Vec` here? The container stores pointers to byte arrays on the heap. Does reducing the number of pointers by one really matter here?

Reply:
Fixed.